### PR TITLE
Expose Dialer into Transport

### PR DESCRIPTION
**Pull request by ernestas-poskus**
@daaku 

Hey, nice package ! (using it in production)

Want to expose/make public Dailer struct into Transport since more or less it is already custom. Reason & benefits is additional configuration of Dialer var 'KeepAlive':

``` go
// KeepAlive specifies the keep-alive period for an active
// network connection.
// If zero, keep-alives are not enabled. Network protocols
// that do not support keep-alives ignore this field.
KeepAlive time.Duration
```

### CODE REVIEW

**Comment by ernestas-poskus** [comment]
Build failed due to go version go1.2.2 linux/amd64

./httpcontrol.go:184: unknown net.Dialer field 'KeepAlive' in struct literal

do you still support go1.2 ?

**Comment by davidbirdsong** [comment]
+1 merge!

**Comment by ernestas-poskus** [comment]
@daaku if pointer (Dialer *net.Dialer) in struct bothers you, I can refactor this to use .Dial() function as standard library does https://golang.org/src/net/http/transport.go:70 (line)

>  Dial func(network, addr string) (net.Conn, error)

**Comment by daaku** [comment]
Apologies for the delays! Yes a `Dial` function would be preferred. And no we don't care about Go 1.2 compatibility, feel free to bump that in `.travis.yml` to get the tests to pass.

**Comment by ernestas-poskus** [comment]
@daaku @davidbirdsong 

Refactored to Dial() function, rebased and bumped go version to 1.3

File: httpcontrol.go [hunk 1]
```diff
@@ -382,6 +399,12 @@ func TransportFlag(name string) *Transport {
 		name+" dial timeout",
 	)
 	flag.DurationVar(
+		&t.DialKeepAlive,
+		name+".dial-keepalive",
+		30*time.Second,
```
Comment (daaku) [inline httpcontrol.go]: We should default this to 0 to not change the behaviour.

**Comment by daaku** [comment]
I'm just going to merge and change that default.

**Comment by daaku** [comment]
Also thanks for your contribution!

File: httpcontrol.go (hunk 1 above)
Comment (ernestas-poskus) [inline httpcontrol.go]: :ok_hand:

**Pull request merged**